README.md

```diff
@@ -18,188 +24,169 @@ The LPS25H should be connected as follows:
 
 ### Constructor
 
-The constructor takes two arguments to instantiate the class: a pre-configured I&sup2;C bus and the sensor’s I&sup2;C address.
+The constructor takes two arguments to instantiate the class: a pre-configured I&sup2;C bus and the sensor’s I&sup2;C address. The I&sup2;C address is optional and defaults to 0xB8 (8-bit address).
 
 ```squirrel
-const LPS25H_ADDR = 0xB8    // 8-bit I2C Address for LPS25H (0x5C on datasheet)
-
-hardware.i2c89.configure(CLOCK_SPEED_400_KHZ)
-pressure <- LPS25H(hardware.i2c89, LPS25H_ADDR)
+const LPS25H_ADDR = 0xBA;    // non-default 8-bit I2C Address for LPS25H (SA0 pulled high)
 
+hardware.i2c89.configure(CLOCK_SPEED_400_KHZ);
+pressureSensor <- LPS25H(hardware.i2c89, LPS25H_ADDR);
 ```
 
 ### Class Methods
 
-### read(*callback*)
-
-The **read()** method reads the pressure in hPa and executes the callback passed as its only parameter with the result. The callback takes a single parameter: the pressure reading (float).
 
-```squirrel
-pressure.read(function(pressureHPa) {
-  server.log(pressureHPa + " hPa")
-})
-```
-
-### getTemp()
+### enable(*state*)
 
-Returns temperature in degrees Celsius.
+Enable (*state* = true) or disable (*state* = false) the LPS25H. The device must be enabled before attempting to read the pressure or temperature.
 
 ```squirrel
-server.log(pressure.getTemp() + "C")
+pressure.enable(true);    // Enable the sensor
 ```
 
-### softReset()
+### read(*callback*)
 
-Reset the LPS25H from software. Device will come up disabled.
+The **read()** method reads the pressure in hPa and executes the callback passed as its only parameter with the result. The callback takes a single parameter, a table. If an error occurs during the reading, the table passed to the callback will contain a key "err", with a description of the error, and the pressure reading will be null. If the pressure is read successfully, it will be stored in the table with the key "pressure".
+
+If the callback is omitted, **read** executes synchronously and returns a table. As with the asynchrounous flow, the "err" key is present in the table if an error occurs. The pressure is stored with the "pressure" key.
```

Comment:
It should, but isn't. Known issue for the time being, with the possibility of refactoring our hardware drivers later on...